### PR TITLE
Updates sample to use a string variant

### PR DESCRIPTION
**Pull request by rossgrambo**
## Purpose
* Changes variant feature flag to demonstrate string usage rather than Boolean
* Small updates due to latest library

## Does this introduce a breaking change?
[x] Yes
[] No

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Follow readme or run locally from sln.

## What to Check
Verify that the following are valid
* azd succeeds and app runs on app service. 
* Greeting should change when logging in with different users. 

## Other Information
To simulate users, use the `/simulate` endpoint.

### CODE REVIEW

File: QuoteOfTheDay/Pages/Simulate.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    public SimulateModel(IVariantFeatureManager featureManager, TelemetryClient telemetryClient)
+    {
+        _featureManager = featureManager;
+        _telemetryClient = telemetryClient;
+    }
+
+    public async void OnGet()
+    {
+        if (HttpContext.Request.Query?["simulate"].Count > 0) {
```
Comment (amerjusupovic) [inline QuoteOfTheDay/Pages/Simulate.cshtml.cs]: I know we don't do it anywhere right now but can someone manually put `?simulate=false` and this would trigger? Either way, we probably could check the value here instead of just checking that `simulate` exists.

Comment (rossgrambo) [reply QuoteOfTheDay/Pages/Simulate.cshtml.cs]: Yeah correct it's a little ugly. My goal was to distinguish between a GET to see the page and a GET that simulates data in the simplest way. POST required supporting xsrf (more code) which also required sending a body (more code) so I gave up on it.

This qeury check means this url would work: `{url}?simulate`

But when writing the url in the javascript I just set it to true because I thought it looked more like people expect for a url. I should probably change it to explicitly check for true.

File: QuoteOfTheDay/Pages/Index.cshtml.cs [hunk 2]
```diff
@@ -40,13 +40,11 @@ public async void OnGet()
...
         {
-            ShowGreeting = variant.Configuration.Get<bool>();
+            Greeting = variant.Configuration?.Get<string>() ?? "";
         }
         else
         {
-            _logger.LogWarning("Greeting variant not found. Please define a variant feature flag in Azure App Configuration named 'Greeting' with 'true' and 'false' variants.");
-
-            ShowGreeting = false;
+            _logger.LogWarning("Greeting variant not found. Please define a variant feature flag in Azure App Configuration named 'Greeting'.");
```
Comment (jimmyca15) [inline QuoteOfTheDay/Pages/Index.cshtml.cs]: Can you define "Greeting" as a local const string that way it can be shared here and in `GetVariantAsync`?

File: QuoteOfTheDay/Pages/Simulate.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public async Task SimulateUser() {
+        string randomUserId = Guid.NewGuid().ToString();
+
+        var activityFeature = HttpContext.Features.Get<IHttpActivityFeature>();
+        activityFeature.Activity.SetBaggage("TargetingId", randomUserId);
+
+        Variant variant = await _featureManager.GetVariantAsync("Greeting", new TargetingContext() { UserId= randomUserId });
+
+        // 50% chance to like for None variant
+        if (variant.Name == "None")
```
Comment (jimmyca15) [inline QuoteOfTheDay/Pages/Simulate.cshtml.cs]: This is dependent upon the variant names. If somebody gets that setup wrong, this seems to do nothing and not warn.

Comment (rossgrambo) [reply QuoteOfTheDay/Pages/Simulate.cshtml.cs]: Removed simulate for now.

File: infra/shared/appConfiguration.bicep [hunk 4]
```diff
@@ -40,29 +40,37 @@ resource variantFeatureFlagGreeting 'Microsoft.AppConfiguration/configurationSto
       "enabled": true,
       "variants": [
         {
-          "name": "On",
-          "configuration_value": true
+          "name": "None"
```
Comment (mrm9084) [inline infra/shared/appConfiguration.bicep]: I know this is a string, but can we avoid None as it's a null in python and it looks confusing like that and we will have it for all languages. How about `No Message`.

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: infra/shared/appConfiguration.bicep (hunk 4 above)
Comment (samsadsam) [reply infra/shared/appConfiguration.bicep]: What about `Default` for this variant and the percentage variant below as well?

File: QuoteOfTheDay/Program.cs [hunk 5]
```diff
@@ -5,14 +5,13 @@
...
-using QuoteOfTheDay;
 
 var builder = WebApplication.CreateBuilder(args);
 
 string defaultConnectionString = "Endpoint=https://azure.azconfig.io;Id=defaultId;Secret=U0VDUkVUX0RVTU1ZX1NUUklORw==";
 
-var appConfigurationConnectionString = Environment.GetEnvironmentVariable("AzureAppConfigurationConnectionString") ?? defaultConnectionString;
-var applicationInsightsConnectionString = Environment.GetEnvironmentVariable("ApplicationInsightsConnectionString") ?? defaultConnectionString;
+var appConfigurationConnectionString = builder.Configuration["AzureAppConfigurationConnectionString"] ?? defaultConnectionString;
+var applicationInsightsConnectionString = builder.Configuration["ApplicationInsightsConnectionString"] ?? defaultConnectionString;
```
Comment (samsadsam) [inline QuoteOfTheDay/Program.cs]: I wonder why this change?

File: QuoteOfTheDay/Program.cs [hunk 6]
```diff
@@ -5,14 +5,13 @@
 using Microsoft.ApplicationInsights.Extensibility;
 using Microsoft.FeatureManagement.Telemetry.ApplicationInsights;
 using Microsoft.FeatureManagement;
-using QuoteOfTheDay;
 
 var builder = WebApplication.CreateBuilder(args);
 
 string defaultConnectionString = "Endpoint=https://azure.azconfig.io;Id=defaultId;Secret=U0VDUkVUX0RVTU1ZX1NUUklORw==";
```
Comment (samsadsam) [inline QuoteOfTheDay/Program.cs]: I know you did not touch this line. Do you think it's safe to move this default connection string into the `appsettings.json` file?

File: infra/shared/appConfiguration.bicep (hunk 4 above)
Comment (zhenlan) [reply infra/shared/appConfiguration.bicep]: `Default` sounds good to me.

File: QuoteOfTheDay/Program.cs (hunk 6 above)
Comment (mrm9084) [inline QuoteOfTheDay/Program.cs]: This endpoint isn't approved as a fake connection string, it looks real. https://review.learn.microsoft.com/en-us/help/platform/reference-sensitive-identifier?branch=main#replacing-guids-in-your-content

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: QuoteOfTheDay/Program.cs (hunk 6 above)
Comment (rossgrambo) [reply QuoteOfTheDay/Program.cs]: This was added to allow aspire to successfully build in order to discover which services it needed to generate. Amer has a note to try and rip this out.

Comment (mrm9084) [reply QuoteOfTheDay/Program.cs]: Can you make it an approved connection string in the mean time.

File: QuoteOfTheDay/Program.cs (hunk 5 above)
Comment (rossgrambo) [reply QuoteOfTheDay/Program.cs]: By using builder.Configuration- it can read UserSecrets. This is the right place to put them anyway and `azd up` populates both environment variables and UserSecrets on start. Additionally- this allows the developer to edit/see them in VS.

File: QuoteOfTheDay/Program.cs (hunk 6 above)
Comment (rossgrambo) [reply QuoteOfTheDay/Program.cs]: That's probably the right place for it- even though it's a fake connection string. I'm not sure if it's safe to move. My understanding is that aspire needs to build this successfully and it's unable to use the builder.Configuration strings. If it's just that `ApplicationInsightsConnectionString` and `AzureAppConfigurationConnectionString` don't exist yet- then I wonder if I should just set those connection strings to this default within appsettings instead. It would clean this up a little

Comment (rossgrambo) [reply QuoteOfTheDay/Program.cs]: It looks like this string is no longer needed (I believe Aspire was removed). Removed it.

Comment (rossgrambo) [reply QuoteOfTheDay/Program.cs]: Removed the string. Builds fine without it.

File: QuoteOfTheDay/Program.cs (hunk 5 above)
Comment (samsadsam) [reply QuoteOfTheDay/Program.cs]: I see, thanks. Resolving

**Review by samsadsam — APPROVED** [review]

**Review by mrm9084 — APPROVED** [review]
